src/pages/VolunteersJobListingFS.page

```diff
@@ -148,9 +151,15 @@
             <apex:outputField value="{!job.Location_Information__c}" />
             </p>
         </apex:outputPanel>
+        
         <apex:outputPanel id="panelJobOnly" rendered="{!job.Number_of_Shifts__c == 0 || ShowShifts == false}" >
-            <input type="button" value="{!$Label.labelButtonSignUp}"  class="cssJobShiftSignup btn" onclick="OpenSignUpDlg(event.clientX, event.clientY, '{!job.Id}', '', '{!JSENCODE(job.Name)}', ''); return false;"/>
-        </apex:outputPanel>         
+            <input type="button" value="{!$Label.labelButtonSignUp}"  class="cssJobShiftSignup btn" onclick="OpenSignUpDlg(event.clientX, event.clientY, '{!job.Id}', '', '{!JSENCODE(job.Name)}', ''); return false;" style="{!IF(job.External_Signup_Url__c == null, '', 'display: none;')}"/>
```

Comment:
rather than setting the style to display:none, couldn't you have just added your logic to the rendered formula in the surrounding outputPanel.  seems like that would be more consistent and readable with the existing code.